teps/0116-referencing-finally-task-results-in-pipeline-results.md

```diff
@@ -0,0 +1,86 @@
+---
+status: proposed
+title: Referencing Finally Task Results in Pipeline Results
+creation-date: '2022-06-24'
+last-updated: '2022-07-15'
+authors:
+- '@@vsinghai'
+---
+
+# TEP-0116: Referencing Finally Task Results in Pipeline Results
+---
+
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Use Cases](#use-cases)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Notes and Caveats](#notes-and-caveats)
+- [Design Details](#design-details)
+- [Design Evaluation](#design-evaluation)
+  - [Reusability](#reusability)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+The `finally` field was introduced to the `PipelineSpec` at the same level as `tasks`. The `finally` field is a list of one or more `PipelineTasks` that guarantee that they will be executed only once all `PipelineTasks` under `tasks` have completed, regardless if that is because of success or error. This proposal improves parity between `finally` field and `tasks` field by supporting referencing `Results` from `finally` in `Pipeline Results`.
+
+## Motivation
+
+Today, `PipelineTasks` in the `Tasks` field can propagate `Results` to `Pipelines` while `PipelineTasks` in the `Finally` cannot propagate `Results` to `Pipelines`. This work was left out of scope when we added support for `Finally` - see [docs](https://github.com/tektoncd/pipeline/blob/main/docs/pipelines.md#cannot-configure-pipeline-result-with-finally) and [issue](https://github.com/tektoncd/pipeline/issues/4923).
```

Comment:
use a permalink because main will change

```suggestion
Today, `PipelineTasks` in the `Tasks` field can propagate `Results` to `Pipelines` while `PipelineTasks` in the `Finally` cannot propagate `Results` to `Pipelines`. This work was left out of scope when we added support for `Finally` - see [docs](https://github.com/tektoncd/pipeline/blob/8a7b0cfa755038f4cfdcd88c314a72a90bcab1a2/docs/pipelines.md#cannot-configure-pipeline-result-with-finally) and [issue](https://github.com/tektoncd/pipeline/issues/4923).
```

Reply:
done